tests/PHPStan/Analyser/data/generics.php

```diff
@@ -344,3 +344,54 @@ public function g() {
 		assertType('T (class PHPStan\Generics\FunctionsAssertType\C, argument)', $a->g());
 	}
 }
+
+/**
+ * class A
+ *
+ * @template T
+ */
+class A {
+	/**
+	 * A::__construct()
+	 *
+	 * @param T $a
+	 */
+	public function __construct($a) {
+	}
+}
+
+/**
+ * class AOfDateTime
+ *
+ * @extends A<\DateTime>
+ */
+class AOfDateTime extends A {
+}
+
+/**
+ * class B
+ *
+ * @template T
+ *
+ * @extends A<T>
+ */
+class B extends A {
+	/**
+	 * B::__construct()
+	 *
+	 * @param T $a
+	 */
+	public function __construc($a) {
+	}
+}
+
+function testClasses() {
+	$a = new A(1);
+	assertType('PHPStan\Generics\FunctionsAssertType\A<int>', $a);
+
+	$a = new AOfDateTime();
+	assertType('PHPStan\Generics\FunctionsAssertType\AOfDateTime', $a);
+
+	$b = new B(1);
+	assertType('PHPStan\Generics\FunctionsAssertType\B<int>', $b);
```

Comment:
These testcases should be enhanced by calling a method some method that `@return T` on these `$a` variables and `$b` variable. Once type resolving in `GenericObjectType::getMethod()` is implemented :)

Reply:
(No action required now.)